api/routes/spa.go

```diff
@@ -16,37 +20,258 @@ type SpaHandler struct {
 	indexPath  string
 }
 
-func NewSpaHandler(staticPath string, indexPath string) SpaHandler {
+func NewSpaHandler(staticPath string, indexPath string) (SpaHandler, error) {
+	staticPathAbs, err := filepath.Abs(staticPath)
+	if err != nil {
+		return SpaHandler{}, fmt.Errorf("static path %s is not valid: %w", staticPath, err)
+	}
+
+	indexPathAbs, err := filepath.Abs(filepath.Join(staticPath, indexPath))
+	if err != nil {
+		return SpaHandler{}, fmt.Errorf("index path %s is not valid: %w", indexPath, err)
+	}
+
+	stat, err := os.Stat(staticPathAbs)
+	if os.IsNotExist(err) {
+		return SpaHandler{}, fmt.Errorf("static path %s does not exist", staticPathAbs)
+	}
+	if os.IsPermission(err) {
+		return SpaHandler{}, fmt.Errorf("no permission to access static path %s", staticPathAbs)
+	}
+	if err != nil {
+		return SpaHandler{}, fmt.Errorf("error accessing static path %s: %w", staticPathAbs, err)
+	}
+	if !stat.IsDir() {
+		return SpaHandler{}, fmt.Errorf("static path %s is not a directory", staticPathAbs)
+	}
+
+	stat, err = os.Stat(indexPathAbs)
+	if os.IsNotExist(err) {
+		return SpaHandler{}, fmt.Errorf("index path %s does not exist", indexPathAbs)
+	}
+	if os.IsPermission(err) {
+		return SpaHandler{}, fmt.Errorf("no permission to access index path %s", indexPathAbs)
+	}
+	if err != nil {
+		return SpaHandler{}, fmt.Errorf("error accessing index path %s: %w", indexPathAbs, err)
+	}
+	if stat.IsDir() {
+		return SpaHandler{}, fmt.Errorf("index path %s is a directory, must be a file", indexPathAbs)
+	}
```

Comment:
How about a new help function like `validDir(dir string) error` and use it to check `indexPath` and `staticPathAbs`? Then the main function could be cleaned as:

```go
func NewSpaHandler(staticPath string, indexPath string) (SpaHandler, error) {
  if err := validDir(staticPath); err != nil {
    return fmt.Errorf("....", staticPath, err)
  }
  if err := validDir(indexPath); err != nil {
    return fmt.Errorf("...", indexPath, err)
  }

  return SpaHandler{
		indexPath:  indexPath,
		staticPath: staticPathAbs,
	}, nil
}
```

BTW, I don't know why you change `staticPath` to `staticPathAbs` but not `indexPath`. I guess both should work properly. For the simplicity, I keep using original value here.